src/Bicep.Core/Emit/TemplateWriter.cs

```diff
@@ -166,6 +166,31 @@ private void EmitParametersIfPresent(JsonTextWriter memoryWriter, ExpressionEmit
             memoryWriter.WriteEndObject();
         }
 
+        private ObjectSyntax EvaluateDecorators(StatementSyntax statement, ObjectSyntax input, TypeSymbol targetType)
```

Comment:
>EvaluateDecorators [](start = 29, length = 18)

Moved the logic to ensure all decorator evaluation semantics are consistent if we ever change them. (If we decide to fix #1712, it'll prevent deviations.)